node/src/blockchain/errors/validation_status.rs

```diff
@@ -12,12 +17,75 @@ pub enum ValidationStatus {
     Reattempting(PreviousAttempts),
 }
 
-#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[derive(Debug, Clone, PartialEq, Eq)]
 pub struct PreviousAttempts {
-    #[serde(flatten)]
     inner: HashMap<BlockchainErrorKind, ErrorStats>,
 }
 
+// I had to implement myself as the default HashMap serialization throws errors if the key value
+// is a compound structure (cannot be projected as a simple string value)
+impl ManualSerialize for PreviousAttempts {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        #[derive(Serialize)]
+        struct Entry<'a> {
+            #[serde(rename = "error")]
+            error_kind: &'a BlockchainErrorKind,
+            #[serde(flatten)]
+            stats: &'a ErrorStats,
+        }
+
+        let mut seq = serializer.serialize_seq(Some(self.inner.len()))?;
+        for (error_kind, stats) in self.inner.iter() {
+            seq.serialize_element(&Entry { error_kind, stats })?;
+        }
+        seq.end()
+    }
+}
+
+impl<'de> ManualDeserialize<'de> for PreviousAttempts {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        deserializer.deserialize_seq(PreviousAttemptsVisitor)
+    }
+}
+
+struct PreviousAttemptsVisitor;
+
+impl<'de> Visitor<'de> for PreviousAttemptsVisitor {
+    type Value = PreviousAttempts;
+
+    fn expecting(&self, formatter: &mut Formatter) -> std::fmt::Result {
+        formatter.write_str("PreviousAttempts")
+    }
+    fn visit_seq<A>(self, mut seq: A) -> Result<Self::Value, A::Error>
+    where
+        A: SeqAccess<'de>,
+    {
+        #[derive(Deserialize)]
+        struct EntryOwned {
+            #[serde(rename = "error")]
+            error_kind: BlockchainErrorKind,
+            #[serde(flatten)]
+            stats: ErrorStats,
+        }
+
+        let mut hash_map: HashMap<BlockchainErrorKind, ErrorStats> = hashmap!();
+        loop {
```

Comment:
Instead of this, you can do it like this:

```rust
       while let Some(entry) = seq.next_element::<EntryOwned>()? {
            error_stats_map.insert(entry.error_kind, entry.stats);
        }

        Ok(PreviousAttempts { inner: error_stats_map })

```